posts/2021-01-29-airports-at-scale.adoc

```diff
@@ -0,0 +1,39 @@
+= Airports at Scale
+nMoncho
+v1.0, 2020-01-29
+:title: Airports at Scale
+:tags: [database, nosql, big data, modelling, programming]
+
+== Assessing NoSQL Databases
+
+In a https://blog.lunatech.com/2018/11/lunatech-airport-assessment/[previous blog post], I explained the assessment we were using at Lunatech, detailing a solution using a common tech stack, Java with Spring Boot and PostgreSQL. An idea mentioned was that it would’ve been possible to also develop a solution using a NoSQL database, since we did receive such solutions, this got me thinking: could every NoSQL family easily solve each requirement? What would be the advantages of using them over SQL? And, more importantly, do they fit in the problem and domain?
+
+In this series of posts I want to explore these questions, where each post focuses on each family.
+
+=== Assessment
+
+You can take a look a the assessment in the previous post, but as a quick refresher, we had to implement four requirements:
+
+* Query Option will ask the user for the country name or code and print the airports & runways at each airport. Bonus point, allow fuzzy search.
+* Report: 10 countries with highest number of airports (with count) and countries with lowest number of airports.
+* Report: Type of runways (as indicated in "surface" column) per country
+* Bonus Report: Print the top 10 most common runway identifications (indicated in "le_ident" column)
+
+=== NoSQL
+
+NoSQL came out as an effort to solve difficulties imposed by traditional relational databases, namely the need to scale to a magnitude not traditionally achieved. Unlike RDMS, NoSQL is actually an umbrella term that groups different implementations such as: document, key-value, column families, graph and so on. Where each offers its own set of features, solving a specific set of problems, but at the same time presenting its own limitations.
```

Comment:
```suggestion
NoSQL came out as an effort to solve difficulties imposed by traditional relational databases, namely the need to scale to a magnitude not traditionally achieved by options like <input them here>. Unlike RDMS, NoSQL is actually an umbrella term that groups different implementations such as: document, key-value, column families, graph and so on. Where each offers its own set of features and solve a specific set of problems, all with their own limitations.
```
This is more of a suggestion. A bit more concise, but the same idea comes across.